packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/session_manager.dart

```diff
@@ -0,0 +1,197 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:amplify_analytics_pinpoint_dart/amplify_analytics_pinpoint_dart.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/sdk/pinpoint.dart';
+import 'package:amplify_core/amplify_core.dart';
+import 'package:intl/intl.dart';
+import 'package:meta/meta.dart';
+
+/// Enum indicating the state of the app session
+enum SessionState {
+  /// App is not active (backgrounded)
+  inactive,
+
+  /// App is running (foregrounded)
+  active,
+
+  /// App is paused
+  paused,
+}
+
+/// Function that listens to Session lifecycle updates
+typedef SessionFunc = void Function(Session);
```

Comment:
Thanks for the docs link, as this pattern is used many times throughout the code I think it makes more sense to have a typedef here to have a single source of truth for that function declaration instead of rewriting throughout. 

Updated name to be better: 
OnSessionUpdated